### PR TITLE
Implement collection details

### PR DESCRIPTION
**Pull request by justmesr**
- [x] Migrate documents to Bootstrap 4
- [x] Attributes tab
- [ ] Events tab (Layed out)
- [x] Link Types tab
- [x] Collection config component

### CODE REVIEW

**Comment by justmesr** [comment]
screenshot from 2017-09-27 12-06-13

screenshot from 2017-09-27 12-06-35

screenshot from 2017-09-27 12-06-42

screenshot from 2017-09-27 12-06-54

screenshot from 2017-09-27 12-07-28

screenshot from 2017-09-27 12-07-57

File: src/main/scripts/src/app/collection/collection-managed-role.guard.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import 'rxjs/add/operator/retry';
+
+@Injectable()
+export class CollectionManageRoleGuard implements CanActivate {
+
+  constructor(private collectionService: CollectionService) {
+  }
+
+  public canActivate(route: ActivatedRouteSnapshot, state: RouterStateSnapshot): Observable<boolean> {
+    const collectionCode = state.url.split('/').reverse()[1];
```
Comment (livthomas) [inline src/main/scripts/src/app/collection/collection-managed-role.guard.ts]: Why don't you get `collectionCode` parameter from `route` here?

File: src/main/scripts/src/app/collection/collection-routing.module.ts [hunk 2]
```diff
@@ -23,13 +23,50 @@ import {RouterModule, Routes} from '@angular/router';
...
+      {
+        path: 'linktypes',
+        component: CollectionLinkTypesComponent
+      },
+      {
+        path: 'events',
+        component: CollectionEventsComponent
+      },
+      {
+        path: 'accessrights',
```
Comment (livthomas) [inline src/main/scripts/src/app/collection/collection-routing.module.ts]: I would use `permissions` in the path.

File: src/main/scripts/src/app/collection/config/access-rights/collection-access-rights.component.html [hunk 3]
```diff
@@ -0,0 +1 @@
+<permissions [resourceType]="collection"></permissions>
```
Comment (livthomas) [inline src/main/scripts/src/app/collection/config/access-rights/collection-access-rights.component.html]: `[resourceType]="'collection'"`

File: src/main/scripts/src/app/collection/config/attribute-list/attribute-list.component.scss [hunk 4]
```diff
@@ -0,0 +1,70 @@
+@import "../../../shared/_lumeer-variables";
+
+:host {
```
Comment (livthomas) [inline src/main/scripts/src/app/collection/config/attribute-list/attribute-list.component.scss]: Why? Angular take care of keeping the separate styles for different components.

File: src/main/scripts/src/app/shared/common.scss [hunk 5]
```diff
@@ -5,3 +5,13 @@ a {
...
 }
+
+ul {
+  list-style-type: none;
+}
+
+input {
+  border: none;
+  background: transparent;
+  outline: none;
```
Comment (livthomas) [inline src/main/scripts/src/app/shared/common.scss]: Do we want these styles in all components?

File: src/main/scripts/src/app/collection/collection-managed-role.guard.ts (hunk 1 above)
Comment (justmesr) [reply src/main/scripts/src/app/collection/collection-managed-role.guard.ts]: I've printed both `route` and `state` objects to console, but can't find the parameter anywhere.. so I've used url instead.

File: src/main/scripts/src/app/collection/config/attribute-list/attribute-list.component.scss (hunk 4 above)
Comment (justmesr) [reply src/main/scripts/src/app/collection/config/attribute-list/attribute-list.component.scss]: It didn't when I used browser for navigating between pages and adding the `:host` fixed the issue.

File: src/main/scripts/src/app/shared/common.scss (hunk 5 above)
Comment (justmesr) [reply src/main/scripts/src/app/shared/common.scss]: Well it didn't break anything and most of the inputs we use are for editable names, (so they should look like regular text) not as much for forms.

Comment (livthomas) [reply src/main/scripts/src/app/shared/common.scss]: I would move these styles to the components where they are really needed. Moreover, try to define a class with these properties and apply it on elements where it is needed. It's better than overriding the default look of elements without styles.

File: src/main/scripts/package.json [hunk 6]
```diff
@@ -74,7 +74,6 @@
     "bootstrap": "4.0.0-beta",
     "lodash": "^4.17.4",
     "minigrid": "^3.1.1",
-    "ng2-webstorage": "^1.5.0",
```
Comment (livthomas) [inline src/main/scripts/package.json]: Please, do not remove this dependency. We will need it.

**Comment by livthomas** [comment]
Please, rebase your changes on `upstream/devel` so we will know if you break the AOT build or not (see [this PR](https://github.com/Lumeer/web-ui/pull/135)).

**Pull request merged**